src/Tetris/PlayerSkeleton.java

```diff
@@ -1,14 +1,69 @@
 package Tetris;
 
+import java.util.*;
+
+import Tetris.Util.Tuple;
+import Tetris.Util.Util;
+
 public class PlayerSkeleton {
 
 	//implement this function to have a working system
 	public int pickMove(State s, int[][] legalMoves) {
-		
 		return 0;
 	}
-	
+
 	public static void main(String[] args) {
+		geneticFunction();
+	}
+
+	// Simulates the replacement of the population by its member's descendants
+	public static void geneticFunction() {
+		ArrayList<Heuristic> population = Util.getRandomHeuristics(Constants.NUMBER_OF_HEURISTICS);
+
+		for (int i = 0; i < Constants.NUMBER_OF_GENERATIONS; i++) {
+			HashMap<Heuristic, Integer> averageScores = getPopulationScores(population);
+			population = generateNextGeneration(averageScores);
```

Comment:
I am confused here, the argument here is `averageScores`, but the method parameter is `population`

Reply:
fixed